src/NexusMods.App.UI/Controls/ModInfo/ModFiles/ModFilesViewModel.cs

```diff
@@ -1,20 +1,111 @@
+using System.Reactive;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using Avalonia.Controls.Selection;
+using DynamicData.Kernel;
 using JetBrains.Annotations;
-using NexusMods.Abstractions.Loadouts;
+using NexusMods.Abstractions.Loadouts.Files;
 using NexusMods.Abstractions.Loadouts.Ids;
-using NexusMods.Abstractions.Loadouts.Mods;
+using NexusMods.App.UI.Controls.Navigation;
 using NexusMods.App.UI.Controls.Trees;
+using NexusMods.App.UI.Controls.Trees.Files;
+using NexusMods.App.UI.Pages.TextEdit;
+using NexusMods.App.UI.Windows;
+using NexusMods.App.UI.WorkspaceSystem;
 using NexusMods.MnemonicDB.Abstractions;
+using ReactiveUI;
+using ReactiveUI.Fody.Helpers;
+using File = NexusMods.Abstractions.Loadouts.Files.File;
 
 namespace NexusMods.App.UI.Controls.ModInfo.ModFiles;
 
 [UsedImplicitly]
-public class ModFilesViewModel(IConnection conn) : AViewModel<IModFilesViewModel>, IModFilesViewModel
+public class ModFilesViewModel : AViewModel<IModFilesViewModel>, IModFilesViewModel
 {
-    public IFileTreeViewModel? FileTreeViewModel { get; private set; }
-    
-    public void Initialize(LoadoutId loadoutId, ModId modId)
+    private readonly IConnection _connection;
+
+    [Reactive] public IFileTreeViewModel? FileTreeViewModel { get; private set; }
+
+    [Reactive] private IFileTreeNodeViewModel? SelectedItem { get; set; }
+
+    public ReactiveCommand<NavigationInformation, Unit> OpenEditorCommand { get; }
+
+    private Optional<LoadoutId> _loadoutId;
+    private Optional<ModId> _modId;
+    private Optional<PageIdBundle> _pageIdBundle;
+
+    public ModFilesViewModel(IWindowManager windowManager, IConnection connection)
     {
-        FileTreeViewModel = new ModFileTreeViewModel(loadoutId, modId, conn);
+        _connection = connection;
+
+        OpenEditorCommand = ReactiveCommand.Create<NavigationInformation>(info =>
+        {
+            var key = SelectedItem!.Key;
+            var targetModId = _modId.Value;
+
+            // NOTE(erri120): not a huge fan of this, but it works
+            var db = connection.Db;
+            var storedFile = db
+                .Find(StoredFile.Hash)
+                .Select(id => db.Get<StoredFile.Model>(id))
+                .FirstOrDefault(storedFile =>
+                {
+                    if (!storedFile.To.Equals(key)) return false;
+                    return targetModId.Equals(storedFile.ModId);
```

Comment:
Check if modId is the same first, should be faster I think if you have lots of files.